src/associations/has-many.js

```diff
@@ -416,24 +404,35 @@ class HasMany extends Association {
    * Un-associate one or several target rows.
    *
    * @param {Model} sourceInstance instance to un associate instances with
-   * @param {Model|Model[]|string|string[]|number|number[]} [targetInstances] Can be an Instance or its primary key, or a mixed array of instances and primary keys
+   * @param {Model|Model[]|string|string[]|number|number[]} [rawTargetInstances] Can be an Instance or its primary key, or a mixed array of instances and primary keys
    * @param {object} [options] Options passed to `target.update`
    *
    * @returns {Promise}
    */
-  async remove(sourceInstance, targetInstances, options = {}) {
-    const update = {
-      [this.foreignKey]: null,
-    };
-
-    targetInstances = this.toInstanceArray(targetInstances);
+  async remove(sourceInstance, rawTargetInstances, options = {}) {
```

Comment:
this should work both with raw instances and non-raw instances, because this method isn't just an internal method, it's the actual `x.removeAssociations(y)` method that get mixed in the model's prototype

```suggestion
  async remove(sourceInstance, targetInstances, options = {}) {
```